### PR TITLE
Event Hubs track 1 receive watchdog

### PR DESCRIPTION
**Pull request by JamesBirdsall**
Add a watchdog on receivers to allow faster recovery if host does not seem to be responding.

### CODE REVIEW

File: sdk/eventhubs/microsoft-azure-eventhubs/src/main/java/com/microsoft/azure/eventhubs/impl/MessagingFactory.java [hunk 1]
```diff
@@ -248,6 +283,100 @@ public void run() {
...
+                    MessagingFactory.this.connection.setCondition(suspect);
+                    MessagingFactory.this.connection.close();
+                    // If the remote host is still responding at the TCP level, then the socket will
+                    // close normally and cleanup/recreation will happen automatically. However, if it
+                    // isn't, then we must call onConnectionError here in order to force cleanup and
+                    // recreation.
+                    try {
+                        Thread.sleep(5000);
+                    } catch (InterruptedException e) {
+                    }
```
Comment (srnagar) [inline sdk/eventhubs/microsoft-azure-eventhubs/src/main/java/com/microsoft/azure/eventhubs/impl/MessagingFactory.java:359]: What happens when the executor is single-threaded? Having a Thread.sleep() on this thread can stall other operations scheduled in this executor.
`executor = Executors.newScheduledThreadPool(1);`

File: sdk/eventhubs/microsoft-azure-eventhubs-eph/pom.xml [hunk 2]
```diff
@@ -14,7 +14,7 @@
   <modelVersion>4.0.0</modelVersion>
   <groupId>com.microsoft.azure</groupId>
   <artifactId>azure-eventhubs-eph</artifactId>
-  <version>3.2.0-beta.1</version> <!-- {x-version-update;com.microsoft.azure:azure-eventhubs-eph;current} -->
+  <version>3.2.0</version> <!-- {x-version-update;com.microsoft.azure:azure-eventhubs-eph;current} -->
```
Comment (conniey) [inline sdk/eventhubs/microsoft-azure-eventhubs-eph/pom.xml:17]: Jim has a script that updates all the version numbers so they are fixed in any other locations where it is used. If you open this file and update to 3.2.0 from -beta.1:
https://github.com/Azure/azure-sdk-for-java/blob/master/eng/versioning/version_data.txt#L27-L29
then run:
`python.exe .\eng\versioning\update_versions.py --update-type library --build-type data --skip-readme`

File: sdk/eventhubs/microsoft-azure-eventhubs/src/main/java/com/microsoft/azure/eventhubs/impl/MessagingFactory.java (hunk 1 above)
Comment (JamesBirdsall) [reply sdk/eventhubs/microsoft-azure-eventhubs/src/main/java/com/microsoft/azure/eventhubs/impl/MessagingFactory.java:359]: A single-threaded executor will not perform adequately for the scenarios that this watchdog is intended for. Anyone who should be using this feature will have a real thread pool. If anyone else is fooling around with the watchdog without consulting us, the worst that can happen is five-second pauses.

File: sdk/eventhubs/microsoft-azure-eventhubs/src/main/java/com/microsoft/azure/eventhubs/EventHubClient.java [hunk 3]
```diff
@@ -70,6 +70,27 @@ static EventHubClient createFromConnectionStringSync(final String connectionStri
...
+     *      {@link ProxyConfiguration#SYSTEM_DEFAULTS} if the system configured proxy settings should be used.
+     * @param maximumSilentTime Use {@link EventHubClientOptions#SILENT_OFF} except on recommendation from the product group.
+     * @return EventHubClient which can be used to create Senders and Receivers to EventHub
+     * @throws EventHubException If Service Bus service encountered problems during connection creation.
+     * @throws IOException       If the underlying Proton-J layer encounter network errors.
+     */
+    static EventHubClient createFromConnectionStringSync(final String connectionString, final RetryPolicy retryPolicy,
+                                                         final ScheduledExecutorService executor,
+                                                         final ProxyConfiguration configuration,
+                                                         final int maximumSilentTime)
```
[the remark this answers is not in the document]
Comment (JamesBirdsall) [reply sdk/eventhubs/microsoft-azure-eventhubs/src/main/java/com/microsoft/azure/eventhubs/EventHubClient.java]: Changed to Duration

File: sdk/eventhubs/microsoft-azure-eventhubs/src/main/java/com/microsoft/azure/eventhubs/EventHubClientOptions.java [hunk 4]
```diff
@@ -92,6 +96,30 @@ public EventHubClientOptions setProxyConfiguration(ProxyConfiguration proxyConfi
...
+    }
+
+    /**
+     * Sets the maximum silent time, in seconds.
+     * Use only on recommendation from the product group.
+     * 
+     * @param maximumSilentTime The time in seconds, or SILENT_OFF. Time must be at least SILENT_MINIMUM_SECONDS.
+     * @return The updated options object.
+     */
+    public EventHubClientOptions setMaximumSilentTime(int maximumSilentTime) {
```
[the remark this answers is not in the document]
Comment (JamesBirdsall) [reply sdk/eventhubs/microsoft-azure-eventhubs/src/main/java/com/microsoft/azure/eventhubs/EventHubClientOptions.java]: Also changed to Duration

File: sdk/eventhubs/microsoft-azure-eventhubs-eph/pom.xml (hunk 2 above)
Comment (JamesBirdsall) [reply sdk/eventhubs/microsoft-azure-eventhubs-eph/pom.xml:17]: Done!

Comment (JimSuplizio) [reply sdk/eventhubs/microsoft-azure-eventhubs-eph/pom.xml:17]: Thank you @conniey for pointing it out and @JamesBirdsall for using it.

**Comment by JamesBirdsall** [comment]
@JimSuplizio I had to make a version change in one more place, spotbugs-aggregate-report/pom.xml, which the python script didn't find. Maybe I ran it incorrectly.

**Comment by JimSuplizio** [comment]
> @JimSuplizio I had to make a version change in one more place, spotbugs-aggregate-report/pom.xml, which the python script didn't find. Maybe I ran it incorrectly.

@JamesBirdsall and I chatted. Windows 10 comes with a python.exe that, when run with no arguments, takes you to the windows store to install python. When you run the same python.exe with arguments, it silently exits and tells you nothing.

**Pull request merged**